core/unit_test/TestSpaceAwareAccessor.hpp

```diff
@@ -0,0 +1,98 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Core.hpp>
+#include <type_traits>
+
+#include <gtest/gtest.h>
+
+template <class T, class ExecutionSpace>
+void test_space_aware_accessor() {
+  using memory_space_t = typename ExecutionSpace::memory_space;
+  using value_type     = std::remove_const_t<T>;
+  Kokkos::View<value_type*, ExecutionSpace> v("V", 100);
+
+  Kokkos::parallel_for(
+      Kokkos::RangePolicy<ExecutionSpace>(0, v.extent(0)),
+      KOKKOS_LAMBDA(int i) { v(i) = i; });
+
+  int errors;
+  using acc_t = Kokkos::Impl::SpaceAwareAccessor<memory_space_t,
+                                                 Kokkos::default_accessor<T>>;
+  acc_t acc{};
+  typename acc_t::data_handle_type ptr = v.data();
+
+  Kokkos::parallel_reduce(
+      Kokkos::RangePolicy<ExecutionSpace>(0, v.extent(0)),
+      KOKKOS_LAMBDA(int i, int& error) {
+        if (acc.access(ptr, i) != ptr[i]) error++;
```

Comment:
very nitpicky but we may want to make an `using acc_base_t = Kokkos::default_accessor<T>` and make this check `acc.access(ptr,i) != acc_base.access(ptr,i)` just to make the requirements clearer

Reply:
I don't agree with that necessarily pretending this tests whether we really just call through seems at best misleading.